crypto/evp/evp_lib.c

```diff
@@ -217,13 +217,10 @@ int EVP_CIPHER_type(const EVP_CIPHER *ctx)
 
 int EVP_CIPHER_block_size(const EVP_CIPHER *cipher)
 {
-    if (cipher->prov != NULL) {
-        if (cipher->blocksize != NULL)
-            return cipher->blocksize();
-        /* We default to a block size of 1 */
-        return 1;
-    }
-    return cipher->block_size;
+    int v = cipher->block_size, ok;
+    ciph_getparams(cipher, &v, sizeof(v), OSSL_CIPHER_PARAM_BLOCK_SIZE,
```

Comment:
many cases of missing blank line after declaration.  see also comment below about turning ```ciph_getparams``` into a function, which makes this be a one-liner.

Reply:
Not sure what you mean re one-liner...

Reply:
```return ciph_getparams(...```

make sense?  sometimes you'll need to use ```?:```

Reply:
It wouldn't quite work right, as `ciph_getparams` (and its function replacement) is designed to handle the provider case only.  Yeah, I know that I can shorten things further, but I also hope that the current form is readable enough.  Would you care to have a look?